### DISCUSSION TITLE
The dynamic API routes deployed on Vercel does not work.

### DISCUSSION THREAD

**Discussion by Hartaithan** [vercel/community#2262] [Help] [answered]
### Update
**WARNING: Links may not work, I removed the test routes from the project.**

### Summary

Everything works locally, but when deploying to Vercel, dynamic API routes return 404

Route path - `/pages/api/test/[slug].ts`

https://trophy-hunt.vercel.app/api/test/example - It doesn't work.
https://trophy-hunt.vercel.app/api/test/[slug] - But it does work.

### Additional information

**Project** - [Link](https://github.com/Hartaithan/trophy-hunt)

**Test Route** - [Link](https://github.com/Hartaithan/trophy-hunt/blob/main/pages/api/test/%5Bslug%5D.ts) 
```
import { type NextApiHandler } from "next";

const handler: NextApiHandler = async (req, res) => {
  return res.status(200).json({ data: req.query });
};

export default handler;
```

**package.json**

```
"dependencies": {
    "@types/node": "18.15.3",
    "@types/react": "18.0.28",
    "@types/react-dom": "18.0.11",
    "eslint-config-next": "13.2.4",
    "next": "13.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
},
```

### Example

https://trophy-hunt.vercel.app/api/test/example

**Comment by amyegan** [comment]
Hi @Hartaithan! I see the error in your example link, but the same code doesn't not produce an error in my test app. It seems something else in your app is leading to the error. I suspect middleware or auth configuration. 

Can you try to reproduce the error in a simplified app? That's usually a good way to isolate the cause.

**Comment by Hartaithan** [comment]
Hi @amyegan!

I published the project on Netlify and everything works fine there.
Maybe I'm wrong, but I don't think the problem is with the app, if locally and on another hosting it works fine.

[https://trophy-hunt.netlify.app/api/test/example](https://trophy-hunt.netlify.app/api/test/example) - It works.
[https://trophy-hunt.netlify.app/api/test/[slug]](https://trophy-hunt.netlify.app/api/test/[slug]) - And this works too.

**Comment by Hartaithan** [comment]
And I tried disabling middleware and nothing changed, I still get 404 from dynamic routes.

**Comment by Hartaithan** [comment]
I think the problem is that Vercel or Next does not recognize a dynamic route name ([slug]) and tries to create a regular static route.
Which is probably confirmed by the fact that this path [https://trophy-hunt.vercel.app/api/test/[slug]](https://trophy-hunt.vercel.app/api/test/%5Bslug%5D)  is available to be called.

**Comment by Hartaithan** [comment, marked as answer]
Oooh, i get it.

I had an `api/API.ts` file in the root of my project, where I kept my axios instance.

And after the build on Vercel, for some reason it appeared in the Output folder as a callable Serverless Function. And as I understood it caused pathname conflicts.

After I moved the `api/API.ts` file to `helpers/api.ts`, the problem was fixed.

Before moving the file: 

<img width="219" alt="Снимок экрана 2023-05-03 в 13 30 15" src="https://user-images.githubusercontent.com/62736094/235833583-e8eac2bf-454b-4f34-a271-529d2497b8ac.png">

After moving the file:

<img width="200" alt="Снимок экрана 2023-05-03 в 13 30 22" src="https://user-images.githubusercontent.com/62736094/235833612-7cfc6c65-ae8f-48d3-8c8c-bacca1963ffd.png">
